linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -807,6 +803,22 @@ private final class Analyzer(config: CommonPhaseConfig,
       }
     }
 
+    private lazy val ancestorsInReflectiveTargetOrder: List[ClassInfo] = {
```

Comment:
It seems we have a trade-off here in how lazy we should be:
We're now always calculating the whole ancestor chain, even if we don't use it. IIUC this depends on how "reflective proxy heavy" a codebase is.

Have you considered using a `Stream` / `LazyList`? IIUC if we get comparable performance on the Scala.js test suite, we're pretty sure it'll be at least as good on less reflective call heavy codebases.